### ISSUE TITLE
ftp-source recursive remote directory

### ISSUE THREAD

**Issue by borg1310** [spring-cloud/stream-applications#583]
hi,

is it possible to configure ftp-soure to recursively traverse the remote directory and transfer all files?

thanks in advance
best regards,
juergen

**Comment by corneil** [comment]
@artembilan may have some input around the spring-ftp-producer.

**Comment by artembilan** [comment]
No, that functionality is not supported yet: https://github.com/spring-projects/spring-integration/issues/3407.

Only the solution available right now is a direct implementation of custom `source` application which would rely on the `FtpOutboundGateway` and its `MGET` command with a `-R` option: https://docs.spring.io/spring-integration/reference/6.3-SNAPSHOT/ftp/outbound-gateway.html#using-the-mget-command

**Comment by borg1310** [comment]
hi @artembilan ,

thanks for your answer.
my use case is to download data from a ftp server that is located in directories sorted by year. So for now I will pursue the approach of using a dynamic remoteDirectoryExpression. (I already tested this and it works. I asked my question anyway because I can't cover all of our use cases with it)

thanks for your help
best regards,

juergen
